File: E-EstateV2/src/app/_interface/costAmount.ts
import { Estate } from "./estate";

export interface CostAmount {
    id: number
    costId: number
    amount: number
    year: number
    estates: Estate[]
    estateId: number
    costSubcategory1Id: number
    costSubcategory1: string
    costSubcategory2: string
    costType: string
    costTypeId: number
    isMature: boolean
    status: string
    createdBy: string
    createdDate: Date
    updatedBy: string
    updatedDate: Date
}

File: E-EstateV2 API/E-EstateV2 API/Models/CostAmount.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace E_EstateV2_API.Models
{
    public class CostAmount
    {
        [Key]
        public int Id { get; set; }
        public decimal amount { get; set; }
        public int year { get; set; }

        public string status { get; set; }
        public string createdBy { get; set; }
        public DateTime createdDate { get; set; }
        public string updatedBy { get; set; }
        public DateTime updatedDate { get; set; }

        [ForeignKey("CostId")]
        public int costId { get; set; }
        public Cost Cost { get; set; }

        [ForeignKey("EstateId")]
        public int estateId { get; set; }
        public Estate Estate { get; set; }
    }
}
